### PR TITLE
ignite-5714 Context switching for pessimistic transactions

### PR DESCRIPTION
**Pull request by voipp**

### CODE REVIEW

**Comment by voipp** [comment]
exec timeout after starting test GridCacheDhtRemoveFailureTest#testPutAndRemovePessimisticTx

**Comment by voipp** [comment]
test org.apache.ignite.internal.processors.cache.distributed.dht.IgniteCacheCrossCacheTxFailoverTest#testCrossCachePessimisticTxFailover fails

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheEntryEx.java [hunk 1]
```diff
@@ -835,15 +835,6 @@ public boolean lockedLocallyByIdOrThread(GridCacheVersion lockVer, long threadId
     @Nullable public GridCacheMvccCandidate candidate(GridCacheVersion ver) throws GridCacheEntryRemovedException;
 
     /**
-     * @param nodeId Node ID.
-     * @param threadId Thread ID.
-     * @return Candidate.
-     * @throws GridCacheEntryRemovedException If entry was removed.
-     */
-    @Nullable public GridCacheMvccCandidate candidate(UUID nodeId, long threadId)
```
Comment (voipp) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheEntryEx.java]: leave it

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMvcc.java [hunk 2]
```diff
@@ -219,26 +219,6 @@ else if (first.owner())
     }
 
     /**
-     *
-     * @param threadId Thread ID.
-     * @param reentry Reentry flag.
-     * @return Local candidate for the thread.
-     */
-    @Nullable private GridCacheMvccCandidate localCandidate(long threadId, boolean reentry) {
```
Comment (voipp) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMvcc.java]: leave it

**Comment by dspavlov** [comment]
IGNITE-5714 | Resolved
-- | --

**Pull request closed without merging**